src/App.tsx

```diff
@@ -1,17 +1,115 @@
-import React from 'react';
+import React, { useState } from 'react';
 import './App.css';
-
+import { TodoList } from './components/TodoList';
+import { PreparedTodo } from './app.typedefs';
+import todos from './api/todos';
 import users from './api/users';
 
+const preparedTodos: PreparedTodo[] = todos.map(todo => ({
+  ...todo,
+  user: users.find(user => user.id === todo.userId) || undefined,
+}));
+
 const App: React.FC = () => {
+  const [todosList, setTodos] = useState(preparedTodos);
+  const [todosTitle, setTodosTitle] = useState('');
+  const [selectedUser, setSelectedUser] = useState(0);
+  const [titleError, setTitleError] = useState(false);
+  const [userError, setUserError] = useState(false);
+
+  const addTodo = (title: string, userId: number) => {
+    const todo: PreparedTodo = {
+      userId,
+      id: todosList.length + 1,
+      title,
+      completed: false,
+      user: users.find(user => user.id === userId),
+    };
+
+    setTodos([...todosList, todo]);
+  };
+
+  const handleSubmit = (event: React.FormEvent):void => {
+    event.preventDefault();
+
+    if (!todosTitle && !selectedUser) {
+      setTitleError(true);
+      setUserError(true);
+    } else if (!selectedUser) {
+      setUserError(true);
+    } else if (!todosTitle) {
+      setTitleError(true);
+    } else {
+      addTodo(todosTitle, selectedUser);
+      setTodosTitle('');
+      setSelectedUser(0);
+    }
```

Comment:
You can replace this long chain of else if statement with a function to validate the input

Reply:
agree. nice choice

Reply:
+

Reply:
I dont get it, where is this long chain of else if?